### PR TITLE
Make a local release of the reference checker

### PR DESCRIPTION
**Pull request by wmdietl**
Contains the changes from #188, merge that first.

This can already be used to run `./gradlew PublishToMavenLocal` and allows local usage.
The new directory `usage-demo` contains a working example that depends on the local release.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -263,3 +333,47 @@ eclipse.classpath {
...
+                licenses {
+                    license {
+                        name = 'The Apache License, Version 2.0'
+                        url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                    }
+                }
+                scm {
+                    connection = 'scm:git:git@github.com:jspecify/jspecify-reference-checker.git'
+                    developerConnection = 'scm:git:git@github.com:jspecify/jspecify-reference-checker.git'
+                    url = 'https://github.com/jspecify/jspecify-reference-checker/'
```
Comment (cpovirk) [inline build.gradle]:
```suggestion
                    url = 'https://github.com/jspecify/jspecify-reference-checker'
```

I hope it doesn't actually matter [here](https://github.com/jspecify/jspecify/pull/578, but conventionally I see these URLs without trailing slashes.

File: usage-demo/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# JSpecify Reference Checker Usage Demo
+
+This is a simple demonstration for how a gradle project can use the JSpecify Reference Checker.
+
+Until the JSpecify Reference Checker is released to Maven Central, on must first run:
+
+````
+./gradlew PublishToMavenLocal
```
Comment (cpovirk) [inline usage-demo/README.md:9]: I would have guessed that _this_ command would need to use `..` and the one on line 14 would need to use `.`. I haven't tested that theory, but does that sound right?

File: usage-demo/build.gradle [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    id 'java'
+    id 'com.diffplug.spotless' version '6.25.0'
+    id 'net.ltgt.errorprone' version '3.1.0'
+    id 'org.checkerframework' version '0.6.42' apply false
+}
+
+ext {
+    versions = [
+        eisopVersion: '3.42.0-eisop4',
+        jspecifyVersion: '0.3.0',
```
Comment (cpovirk) [inline usage-demo/build.gradle]:
```suggestion
        jspecifyVersion: '1.0.0',
```

(🎉 as usual)

**Review by cpovirk — APPROVED** [review]

File: usage-demo/README.md (hunk 2 above)
Comment (wmdietl) [reply usage-demo/README.md:9]: I think the number of dots is correct. I added more explanations to hopefully make it clearer.

**Review by wmdietl — COMMENTED** [review]
Thanks for the review!

File: usage-demo/README.md (hunk 2 above)
Comment (cpovirk) [reply usage-demo/README.md:9]: Yes, thank you, that completely clarified things for me.

**Pull request merged**